### PR TITLE
monitoring: use persistent storage for Prometheus

### LINKED ISSUE #1515: Update Prometheus deployment to use a PVC
**Component**:

salt, monitoring, tests, documentation

**Why this is needed**:

In order to have persistent storage for the monitoring stack, we need Prometheus to us a PV for its storage.

**What should be done**:

- update the Prometheus deployment to come with a PVC (which size do we request?)
- update the E2E tests: between the bootstrap and the execution of the E2E test we need to create a PV that can satisfy the PVC (otherwise Prometheus won't start and lot of things won't work…)
- update the documentation to inform the users that they should create a PV for Prometheus after the bootstraping.

**Test plan**:

Check that Prometheus does use persistent storage (data should be preserved accross pod restart).

### PR DESCRIPTION
**Pull request by NicolasT**
These changes configure the Prometheus stateful services (i.e.,
Prometheus and AlertManager) to use a persistent volume for storage
instead of ephemeral `emptyDir` volumes.

As a result, these services don't 'work' out of the box after initial
deployment. We add some example `Volume` templates using the
`sparseLoopDevice` storage type to create proper volumes on a node after
deployment, which should allow the services to start.

Furthermore, the Vagrant deployment and CI are extended to create the
required `Volume`s such that the Prometheus services *are* available
after bootstrapping the system.

Note: during an earlier implementation of this feature (#1523), we ran
into a dependency issue between Prometheus deployment and the MetalK8s
UI deployment, which requires the first to be available. Instead, given
this PR, we use a hard-coded `NodePort` port number to access
Prometheus, not requiring the service to be available during bootstrap,
and point the UI to the workload-plane IP of the bootstrap node. This
ought to be improved once we use an `Ingress` to access the UI.

Fixes: #1515
See: https://github.com/scality/metalk8s/issues/1515
See: https://github.com/scality/metalk8s/pull/1523

### CODE REVIEW

**Comment by NicolasT** [comment]
@bert-e wait

**Comment by NicolasT** [comment]
Once #1711 is in, this could be picked up again.

**Comment by NicolasT** [comment]
@bert-e wait

**Comment by NicolasT** [comment]
TODOs:

- [x] Docs

File: eve/create-volumes.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    fi
+    sleep 1
+done
+[ $OK -eq 1 ] || (echo "AlertManager not Running"; false)
+
+OK=0
+echo 'Waiting for Prometheus to be running'
+for _ in $(seq 1 60); do
+    PHASE=$(kubectl -n metalk8s-monitoring get pod prometheus-prometheus-operator-prometheus-0 -o jsonpath="{.status.phase}")
+    if [ "x${PHASE}" = "xRunning" ]; then
```
Comment (slaperche-scality) [inline eve/create-volumes.sh:66]: why the `x`?

File: eve/create-volumes.sh [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        OK=1
+        break
+    fi
+    sleep 1
+done
+[ $OK -eq 1 ] || (echo "Prometheus not Running"; false)
```
Comment (slaperche-scality) [inline eve/create-volumes.sh:72]: I guess the `wait for PV` and `wait for Pod` could be in function, and then we would call each of them twice, but meh it's shell scripts anyway…

File: salt/metalk8s/addons/prometheus-operator/deployed/storageclass.sls [hunk 3]
```diff
@@ -1,14 +1,18 @@
-#!jinja | kubernetes kubeconfig=/etc/kubernetes/admin.conf&context=kubernetes-admin@kubernetes
+#! kubernetes kubeconfig=/etc/kubernetes/admin.conf&context=kubernetes-admin@kubernetes
```
Comment (slaperche-scality) [inline salt/metalk8s/addons/prometheus-operator/deployed/storageclass.sls:1]: the move (and update) could have been in a dedicated commit, to explain the rationale behind the change/move.

Could also edit the current commit message to explain it.

File: eve/create-volumes.sh (hunk 1 above)
Comment (NicolasT) [reply eve/create-volumes.sh:66]: Old habits don't go away easily... https://unix.stackexchange.com/a/136632

File: eve/create-volumes.sh (hunk 2 above)
Comment (NicolasT) [reply eve/create-volumes.sh:72]: Not worth the trouble IMHO.

File: salt/metalk8s/addons/prometheus-operator/deployed/storageclass.sls (hunk 3 above)
Comment (NicolasT) [reply salt/metalk8s/addons/prometheus-operator/deployed/storageclass.sls:1]: I'll amend the commit message. Not fond of doing moves like this (which don't bring any value per-se outside of the context of the actual commit doing the PV stuff and such) in separate commits.

File: eve/create-volumes.sh (hunk 1 above)
Comment (slaperche-scality) [reply eve/create-volumes.sh:66]: Ha good to know, thanks for the pointer :slightly_smiling_face:

File: eve/create-volumes.sh (hunk 2 above)
Comment (slaperche-scality) [reply eve/create-volumes.sh:72]: Yeah, especially if we are going to rewrite this in Python at some point :stuck_out_tongue:

**Review by slaperche-scality — APPROVED** [review]

**Pull request merged**